tests/main/views/test_direct_award.py

```diff
@@ -350,6 +363,122 @@ def test_overview_renders_specific_elements_for_search_downloaded_state(self):
 
+    @pytest.mark.parametrize(
+        ('framework_status', 'following_framework_status', 'locked_at', 'position', 'heading'),
+        (
+            ('live', 'coming', None, 'sidebar',
+                'G-Cloud\xa010 services will replace existing services on Wednesday 5 January 2000'),
+            ('live', 'open', None, 'sidebar',
+                'G-Cloud\xa010 services will replace existing services on Wednesday 5 January 2000'),
+            ('live', 'pending', None, 'sidebar',
+                'G-Cloud\xa010 services will replace existing services on Wednesday 5 January 2000'),
+            ('live', 'standstill', None, 'sidebar',
+                'G-Cloud\xa010 services will replace existing services on Wednesday 5 January 2000'),
+            ('live', 'live', None, 'sidebar',
+                'G-Cloud\xa010 services replaced G-Cloud\xa09 services on Wednesday 5 January 2000'),
+            ('expired', 'live', None, 'sidebar',
+                'G-Cloud\xa010 services replaced G-Cloud\xa09 services on Wednesday 5 January 2000'),
+            ('live', 'coming', '2018-06-25T21:57:00.881261Z', 'banner',
+                'The G-Cloud\xa09 services you found will expire soon'),
+            ('live', 'open', '2018-06-25T21:57:00.881261Z', 'banner',
+                'The G-Cloud\xa09 services you found will expire soon'),
+            ('live', 'pending', '2018-06-25T21:57:00.881261Z', 'banner',
+                'The G-Cloud\xa09 services you found will expire soon'),
+            ('live', 'standstill', '2018-06-25T21:57:00.881261Z', 'banner',
+                'The G-Cloud\xa09 services you found will expire soon'),
+            ('live', 'live', '2018-06-25T21:57:00.881261Z', 'banner',
+                'The G-Cloud\xa09 services you found will expire soon'),
+            ('expired', 'live', '2018-06-25T21:57:00.881261Z', 'banner',
+                'The G-Cloud\xa09 services you found have expired'),
+        )
+    )
+    @mock.patch('app.main.views.g_cloud.content_loader')
+    def test_overview_displays_correct_temporary_message(
+        self, content_loader_mock, framework_status, following_framework_status, locked_at, position, heading
+    ):
+        self._set_project_locked_and_framework_states(locked_at, framework_status, following_framework_status)
+        content_loader_mock.get_metadata.return_value = 'g-cloud-10'
+
+        res = self.client.get('/buyers/direct-award/g-cloud/projects/1')
+        assert res.status_code == 200
+
+        body = res.get_data(as_text=True)
+        doc = html.fromstring(body)
+
+        if position == 'sidebar':
+            assert len(doc.xpath("//div[@class='temporary-message']")) == 1
+            assert not doc.xpath("//div[@class='temporary-message-banner']")
+        elif position == 'banner':
+            assert len(doc.xpath("//div[@class='temporary-message-banner']")) == 1
+            assert not doc.xpath("//div[@class='temporary-message']")
+
```

Comment:
Maybe an `else raise` to catch a misconfigured parameter?

Reply:
sure